docker-compose.yml

```diff
@@ -0,0 +1,83 @@
+version: '3.8'
+
+services:
+  db:
+    image: 'mysql:9.0.1'
+    restart: always
+    working_dir: /app
+    environment:
+      MYSQL_ROOT_PASSWORD: ${MYSQL_ROOT_PASSWORD}
+      MYSQL_DATABASE: ${MYSQL_DATABASE}
+      MYSQL_USER: ${MYSQL_USER}
+      MYSQL_PASSWORD: ${MYSQL_PASSWORD}
+    volumes:
+      - db_data:/var/lib/mysql
+    networks:
+      - internal
+  api:
+    build:
+      context: ./backend
+      dockerfile: Dockerfile
+    command: >
+      sh -c "sleep 30 && npm run prisma-up && npm run prisma-gen && npm run start"
+    depends_on:
+      - db
+    environment:
+      DATABASE_URL: "mysql://root:${MYSQL_ROOT_PASSWORD}@db:3306/${MYSQL_DATABASE}"
+      NODE_ENV: production
+      HEX_USER_ID_OFFSET: 0
+      PORT: 3000
+      SMTP_PORT: 465
+      SMTP_HOST: ${SMTP_HOST}
+      SMTP_USERNAME: ${SMTP_USERNAME}
+      SMTP_PASSWORD: ${SMTP_PASSWORD}
+      SMTP_SENDER: ${SMTP_SENDER}
+      JWT_SECRET: ${JWT_SECRET}
+      RECAPTCHA_SECRET_KEY: ${RECAPTCHA_SECRET_KEY}
+      SIGN_IN_URL: ${PUBLIC_FRONTEND_URL}/auth/login
+      RESET_PASSWORD_URL: ${PUBLIC_FRONTEND_URL}/auth/password
+      CORS_ORIGIN: ${PUBLIC_FRONTEND_URL}
+      GOOGLE_AUTH_CLIENT_ID: none
```

Comment:
Are there inconsistencies with the env variables here?

MYSQL_USER is available as a var but set as "root" in the connection URL.
SMTP_PORT and some other settings are not exposed as env vars for the docker. Is there a reason for this?

If we don't expose the MySQL server outside of docker, then maybe we should just set all values as constants to reduce the effort to set this up. Not sure if that is a good practice tho - open to suggestions.

Reply:
One thing I noticed too is the frontend-based environment variables, but apparently by own testing, they only get loaded through the .env file, not being set in the environment.

Reply:
Maybe we set default values for all the MySQL env variables and add a comment to the `.env.docker.example` file that these variables are optional. That way it is clear to the user that they exist and can be changed, but setup is easy

Reply:
> If we don't expose the MySQL server outside of docker, then maybe we should just set all values as constants to reduce the effort to set this up. Not sure if that is a good practice tho - open to suggestions.

I would suggest not allowing the change of the DATABASE_URL (especially since only mysql is supported as the database), but providing the MYSQL_DATABASE etc from the environment file and through the documentation stating that these should be kept default if not using an external one.

Reply:
Done, added a workaround for the Environment variables

Reply:
@DevLeoko initially thats what i did only relevant things being variable but it was commented out earlier to

Reply:
Updated to be included in a file for docker-based environments